indico/modules/events/registration/util.py

```diff
@@ -995,3 +997,24 @@ def get_persons(registrations, include_accompanying_persons=False):
                             'registration': registration,
                             'is_accompanying': True} for person in registration.accompanying_persons)
     return persons
+
+
+def resize_uploaded_registration_picture(source, size):
+    """Resize uploaded registration picture to the given size."""
+    try:
+        picture = Image.open(source)
+    except OSError:
+        return None
+    picture = ImageOps.exif_transpose(picture)
+    size_x, size_y = picture.size
+    if max(size_x, size_y) < size:
+        source.seek(0)
+        return source
+    ratio = size / max(size_x, size_y)
+    picture = picture.resize((int(ratio * size_x), int(ratio * size_y)), Image.Resampling.BICUBIC)
+    image_bytes = BytesIO()
+    if picture.mode != 'RGB':
+        picture = picture.convert('RGB')
+    picture.save(image_bytes, 'JPEG')
```

Comment:
Wouldn't it make sense to unconditionally save uploaded registration pictures as JPEG? My rationale is the following:
1. Avoiding to store heavier than necessary pictures (PNG is less compressed than JPEG).
2. Consistency (resized pictures will be stored in JPEG anyways, why not making sure all of them are?).

Reply:
yes, absolutely (just like we convert user avatars to PNGs)